packages/trees/backend/sirius-components-collaborative-trees/src/main/java/org/eclipse/sirius/components/collaborative/trees/handlers/TreePathEventHandler.java

```diff
@@ -68,17 +74,28 @@ public void handle(One<IPayload> payloadSink, Many<ChangeDescription> changeDesc
         if (treeInput instanceof TreePathInput input) {
             Optional<ITreePathProvider> optionalPathProvider = this.treePathProviders.stream().filter(treePathProvider -> treePathProvider.canHandle(tree)).findFirst();
             if (optionalPathProvider.isPresent()) {
-                IPayload resultPayload = optionalPathProvider.get().handle(editingContext, tree, input);
-                if (resultPayload instanceof TreePathSuccessPayload) {
-                    payload = resultPayload;
-                } else if (resultPayload instanceof ErrorPayload errorPayload) {
-                    this.logger.warn(errorPayload.messages().stream().map(Message::body).collect(Collectors.joining("; ")));
-                }
+                payload = optionalPathProvider.get().handle(editingContext, tree, input);
+            } else {
+                payload = this.handleDefaultTreePath(editingContext, tree, input);
```

Comment:
Everything will be delegated to the description so their won't be a fallback

Reply:
We need to keep the fallback mechanism in place to handle special cases.
For instance, to avoid expanding special tree items (virtual ones) that are irrelevant to the Expand all action.
Another example could be handling infinite loops differently (for instance avoiding to loop at all).